src/mudlet.cpp

```diff
@@ -4122,6 +4122,553 @@ void mudlet::playSound(const QString& s, int soundVolume)
+                    file.flush();
+                    file.close();
+
+                    reply->deleteLater();
+                }, getReply));
+
+            wasFileDownloaded = true;
+        }
+    }
+
+    return wasFileDownloaded;
+}
+
+QMediaPlayer* mudlet::getMSPSoundMediaPlayer()
+{
+    QMediaPlayer* pPlayer = nullptr;
+    QPointer<Host> pHost = getActiveHost();
+
+    if (pHost) {
+        QListIterator<QMediaPlayer*> itMusicBox2(mMusicBoxList);
+
+        while (itMusicBox2.hasNext()) { // Find first available inactive QMediaPlayer
+            QMediaPlayer* pTestPlayer = itMusicBox2.next();
+
+            if (pTestPlayer->state() != QMediaPlayer::PlayingState && pTestPlayer->mediaStatus() != QMediaPlayer::LoadingMedia) {
+                pPlayer = pTestPlayer;
+                break;
+            }
+        }
+
+        if (!pPlayer) { // No available QMediaPlayer, create a new one.
+            pPlayer = new QMediaPlayer(this);
+
+            if (!pPlayer) { // It should be impossible to ever reach this.
+                TDebug(QColor(Qt::white), QColor(Qt::red)) << QStringLiteral("mudlet::getMSPMediaPlayer() WARNING - Unable to create new QMediaPlayer object\n") >> 0;
+                return pPlayer;
+            }
+
+            mMusicBoxList.append(pPlayer);
+        }
+
+        // Remove any previous connection to the signal of this QMediaPlayer,
+        // theoretically this might be movable to be within the lambda function of
+        // the following connect(...) but that does seem a bit twisty and this works
+        // well enough!
+        disconnect(pPlayer, &QMediaPlayer::stateChanged, nullptr, nullptr);
+
+        connect(pPlayer, &QMediaPlayer::stateChanged, [=](QMediaPlayer::State state) {
+            if (state == QMediaPlayer::StoppedState) {
+                TEvent soundFinished {};
+                soundFinished.mArgumentList.append("sysSoundFinished");
+                soundFinished.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+                soundFinished.mArgumentList.append(pPlayer->media().canonicalUrl().fileName());
+                soundFinished.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+                soundFinished.mArgumentList.append(pPlayer->media().canonicalUrl().path());
+                soundFinished.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+                if (pHost) {
+                    // The host may have gone away if the sound was a long one
+                    // and we are multi-playing so we ought to test it...
+                    pHost->raiseEvent(soundFinished);
+                }
+            }
+        });
+    }
+
+    return pPlayer;
+}
+
+void mudlet::playMSPSound(QString& soundFileName, int soundVolume, int soundLength, int soundPriority, const QString& soundType, const QString& soundUrl)
+{
+    QPointer<Host> pHost = getActiveHost();
+
+    if (!pHost) {
+        return;
+    }
+
+    soundFileName.replace('\\', "/");
```

Comment:
Shouldn't `"/"` be `/` - it, too, is a *single* character - not a string - and presumably a one-for-one character replacement is marginally faster/easier than replacement of one character by a string which could potentially (in other situations) occupy a different amount of memory... oh, and to avoid the whole not being wrapped for translation/not being translated issue I would change this to `soundFileName.replace(QLatin1Char('\\'), QLatin1Char('/');`

Reply:
Resolving.